heclib/heclib_f/src/zsitsi6.f

```diff
@@ -228,6 +246,150 @@ SUBROUTINE zsitsi6(IFLTAB, CPATHNAME, ITIMES, SVALUES, DVALUES,
          IF (JSTAT.NE.0) GO TO 940
           CALL zsetfi (IFLTAB, 'PSEU', CPATH1(1:NPATH), INTLPS, IST)
       ENDIF
+      !-----------------------------------------------!
+      ! convert to native vertical datum if necessary !
+      !-----------------------------------------------!
+      effectiveNuhead = nuhead
+      effectiveIuhead = 0
+      do i = 1, min(size(effectiveIuhead), nuhead)
+        effectiveIuhead(i) = iuhead(i)
+      end do
+      cc = cpath(ibpart(3):iepart(3))
+      call upcase(cc)
+      if (index(cc,'ELEV').eq.1) then
+        !-----------------------!
+        ! elevation time series !
+        !-----------------------!
+        if (nuhead.eq.0) then
```

Comment:
Make sure zsrtsi6 and zspdi6 both have this logic (using user header from disk if one isn't specified in call)